app/controllers/developers_controller.rb

```diff
@@ -41,7 +41,18 @@ def update
   end
 
   def show
-    @developer = find_developer!
+    if Feature.enabled?(:obfuscate_developer_urls, user: nil)
+      # Checks if :id contains non-Integer characters (hashid)
+      if /\D+/.match?(params[:id])
+        @developer = Developer.find_by_hashid!(params[:id])
+      else
+        @developer = Developer.find(params[:id])
+        redirect_to @developer, status: 302, notice: t(".redirection", url: developer_url(@developer))
+      end
+    else
+      @developer = Developer.find(params[:id])
+    end
+
```

Comment:
I think we need some tests for this. There are a few edge cases that aren't covered and I'm having some trouble reasoning them about.

Maybe pulling it out to a `DeveloperFinder` object, or similar, could help make that a bit easier?

Reply:
I extracted the logic to a `Developers::Finder` object and added some [tests for it](https://github.com/joemasilotti/railsdevs.com/pull/667/files#diff-39d8d4541629ba3c175e9180086b2d3c4291153cab51512fb185d0d1a2336878R1-R20).

For the task:
> 302 redirect /developers/:id to /developers/:hashid with a flash message

I modified the [tests here](https://github.com/joemasilotti/railsdevs.com/pull/667/files#diff-d18d5d631cf8cfd340082086aca016edb6da051bb625499cd2d6a0ee1874a3d3R60-R65) in `test/integration/developers_test.rb`